src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -369,16 +372,15 @@ const PlrDaoStakingTransactionBlock = ({
   );
 
   const getNftList = async () => {
+    if (!accountAddress || !providerAddress || !sdk) return Promise.resolve();
     try {
-      if (!accountAddress || !providerAddress || !sdk) return;
-      const output = await sdk.getNftList({
-        account: accountAddress || providerAddress,
-      });
-      let hasNFTContractAddress = output?.items?.filter((nft) => nft.contractAddress === plrDaoMemberNFT.address);
-      if (hasNFTContractAddress?.length) {
-        setIsNFTMember(true);
-      }
-    } catch (error) {
+      const nftCollection = await Promise.all(
+        [providerAddress, accountAddress].map(
+          async (address) => await getNftsForChainId(CHAIN_ID.POLYGON, address, true)
+        )
+      ).then(([provider, account]) => provider.concat(account));
```

Comment:
PLEASE PLEASE PLEASE never mix `await` with `then` and never mix `await` with `.catch`

also please if it's a promise function u don't need to return additional await

this is much easier to read and fixes the above:

```suggestion
      const [providerAddressNfts, accountAddressNfts] = await Promise.all([providerAddress, accountAddress].map((address) => getNftsForChainId(CHAIN_ID.POLYGON, address, true)));
      const nftCollection = [...providerAddressNfts, ...accountAddressNfts];
```